app/components/Field/index.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+import * as React from 'react';
+import isObject from 'utils/isObject';
+import consumeContextBroadcast from 'utils/consumeContextBroadcast';
+import type { FormData } from 'components/Form';
+
+type FieldProps = {
+  component: React.ElementType,
+  name: string,
+  handleRef: (ref: ?React.ElementRef<any>) => void,
+  formData: FormData,
+};
+
+/**
+ * `Field` component.
+ *
+ * Used to connect any form field to the form data generated by the Form component.
+ *
+ * The prop `component` can be a Component, a stateless function component, or
+ * a string for DOM form fields (`input`, `select`). This is the component that will
+ * be rendered.
+ */
+export class Field extends React.Component<FieldProps> {
+  static defaultProps = {
+    component: 'input',
+    handleRef: null,
+  };
+
+  /**
+   * Return a field value from a SyntheticEvent or a value
+   */
+  getValue = (eventOrValue: any) => {
+    if (!isObject(eventOrValue)) return eventOrValue;
+
+    const target = eventOrValue.target;
+    if (target) {
+      const type = target.type;
+      if (type === 'checkbox') {
+        return target.checked || '';
+      }
+      return target.value;
+    }
+    return eventOrValue;
+  };
+
+  getFieldData = () => {
+    const { name, formData } = this.props;
+
+    return formData.fields[name];
+  };
+
+  /**
+   * Handle change from the underlying component
+   */
+  handleChange = (eventOrValue: SyntheticEvent<HTMLInputElement> | mixed) => {
```

Comment:
this argument should be the same type as the one in `getValue`. How about creating a named type and assigning it to both places?

```js
type UnkownValue = SyntheticEvent<HTMLInputElement> | mixed;
````